spec/controllers/spree/admin/orders/customer_details_controller_spec.rb

```diff
@@ -51,22 +51,22 @@
       end
 
       context "when adding details of a registered user" do
-        it "redirects to shipments on success" do
+        it "redirects to payments on success" do
           spree_post :update, order: { email: user.email, bill_address_attributes: address_params, ship_address_attributes: address_params }, order_id: order.number
 
           order.reload
 
-          expect(response).to redirect_to spree.admin_order_customer_path(order)
+          expect(response).to redirect_to spree.new_admin_order_payment_path(order)
```

Comment:
This redirect will be a sensible outcome if the user is placing a new order and entering the customer details for the first time... but what if they're editing customer details on an order that's already compete and already has completed payments?

Reply:
Yeah I think you're right - I might have been trying to do a little too much here. It would be a nice UX if we automatically moved you through the screens in order, but I think changing the flow as little as possible is probably best here.